lib/inspec/plugin/v2/installer.rb

```diff
@@ -342,13 +345,17 @@ def install_gem_to_plugins_dir(new_plugin_dependency, # rubocop: disable Metrics
 
       # Activate all current plugins before trying to activate the new one
       loader.list_managed_gems.each do |spec|
-        next if spec.name == new_plugin_dependency.name && update_mode
+        # Skip in case of update mode
+        # Skip in case using a resource pack plugin
+        next if spec.name == new_plugin_dependency.name && (update_mode || is_resource_pack_gem?(new_plugin_dependency.name))
```

Comment:
can we rather check for spec name and version here?

Reply:
So the new plugin dependency version is going to be an updated version and it will never match the version of old cached dependencies. 
We can clear all old cache safely because in this scenario a user has made a decision to either upgrade and downgrade and the old cache is not required anyway. There is no reason to have a version check. Do you have any specific reasoning behind having a version check?

Reply:
okay it is not the case that the user decided to upgrade or downgrade. There is a scenario where the user simply wants to run with different versions.